server/sonar-web/src/main/js/apps/permissions/project/components/PageHeader.tsx

```diff
@@ -17,43 +17,48 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-// @flow
-import React from 'react';
+import * as React from 'react';
+import ApplyTemplate from './ApplyTemplate';
+import { Component } from '../../../../app/types';
 import { translate } from '../../../../helpers/l10n';
-import ApplyTemplateView from '../views/ApplyTemplateView';
 
-/*::
-type Props = {|
-  component: {
-    configuration?: {
-      canApplyPermissionTemplate: boolean,
-      canUpdateProjectVisibilityToPrivate: boolean
-    },
-    key: string,
-    qualifier: string,
-    visibility: string
-  },
-  loadHolders: () => void,
-  loading: boolean
-|};
-*/
+interface Props {
+  component: Component;
+  loadHolders: () => void;
+  loading: boolean;
+}
+
+interface State {
+  applyTemplateModal: boolean;
+}
+
+export default class PageHeader extends React.PureComponent<Props, State> {
+  mounted = false;
+  state: State = { applyTemplateModal: false };
+
+  componentDidMount() {
+    this.mounted = true;
+  }
 
-export default class PageHeader extends React.PureComponent {
-  /*:: props: Props; */
+  componentWillUnmount() {
+    this.mounted = false;
+  }
 
-  handleApplyTemplate = (e /*: Event & { target: HTMLButtonElement } */) => {
-    e.preventDefault();
-    e.target.blur();
-    const { component, loadHolders } = this.props;
-    const organization = component.organization ? { key: component.organization } : null;
-    new ApplyTemplateView({ project: component, organization })
-      .on('done', () => loadHolders())
-      .render();
+  handleApplyTemplate = (event: React.SyntheticEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.currentTarget.blur();
```

Comment:
I think those 2 lines can be removed

Reply:
I try to always prevent the default behavior (just in case) and remove the focus from a link/button

Reply:
mmh I think we are not very consistant on this then. We are for links but not for buttons. We can find a lot of buttons handler with and without the default behavior.
Let's say we always prevent it from now on.
Then maybe we should have a button component that always does it ?

Reply:
good idea